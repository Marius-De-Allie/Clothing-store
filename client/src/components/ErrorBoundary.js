import React from 'react';

class ErrorBoundary extends React.Component {
    // component state.
    state = {
        hasErrored: false
    }

    static getDerivedStateFromError(error) {
        // set component hasErrored state to true
        return {hasErrored: true};
    }

    componentDidCatch(error, info) {
        console.log(error)
    }

    render() {
        const { hasErrored } = this.props;
        return hasErrored ? <div>Something went wrong</div> : 
            this.props.children
    }
};

export default ErrorBoundary;