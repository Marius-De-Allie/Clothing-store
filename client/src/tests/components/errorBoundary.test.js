import React from 'react';
import { shallow } from 'enzyme';
import ErrorBoundary from '../../components/ErrorBoundary';

describe('ErrorBoundary component', () => {
    it('renders properly', () => {
        const wrapper = shallow(<ErrorBoundary />);
        expect(wrapper).toMatchSnapshot();
    });
});