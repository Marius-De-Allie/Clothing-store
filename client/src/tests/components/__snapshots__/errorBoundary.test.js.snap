// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component renders properly 1`] = `ShallowWrapper {}`;
